### PR TITLE
Add vector search example

### PR DESCRIPTION
**Pull request by hmarszalek**
Vector search needs nothing new from this driver: the CQL vector type is already supported, and `ORDER BY ... ANN OF ?` is plain CQL sent through the normal `execute()` path. This PR proves it with a minimal working example.

`examples/vector/vector-search-ann.js` stores comment embeddings in a `vector<float, 3>` column, creates a vector index, and runs an ANN similarity query.
`examples/vector/vector-insert-select.js` covers plain vector column insert/select.

Enabled tests for map, set, ascii, and varint variants inside a vector.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- ~[ ] I added relevant tests for new features and bug fixes.~
- [x] All commits compile, pass static checks and pass tests.
- [x] PR description sums up the changes and reasons why they should be introduced.
- ~[ ] I have provided docstrings for the public items that I want to introduce.~
- ~[ ] I have made sure that the type stubs / TS definitions match the JS public API.~
- ~[ ] I have adjusted the documentation in `./docs/source/`.~
- ~[ ] I have adjusted the migration documentation (removed/changed API) in `./docs/source/migration-guide`.~
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

File: examples/vector/vector-search-ann.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                prepare: true,
+            });
+            if (result.rows.length > 0) {
+                return result.rows;
+            }
+        } catch (_) {
+            // Vector search not yet ready; wait and retry.
+        }
+        if (Date.now() > deadline) {
+            return [];
```
[the remark this answers is not in the document]
Comment (hmarszalek) [reply examples/vector/vector-search-ann.js]: I am not sure how to filter those errors in an elegant way (not string parsing).

File: examples/vector/vector-insert-select.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    await client.connect();
+
+    await client.execute(
+        "CREATE KEYSPACE IF NOT EXISTS examples WITH replication =" +
+            "{'class': 'NetworkTopologyStrategy', 'replication_factor': '1' }",
+    );
+
+    await client.execute(
+        "CREATE TABLE IF NOT EXISTS examples.vector_comments " +
+            "(id uuid PRIMARY KEY, comment text, comment_vector vector<float, 5>)",
```
Comment (QuerthDP) [inline examples/vector/vector-insert-select.js]: PR cover letter says `vector<float, 3>`

File: examples/vector/vector-insert-select.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    );
+
+    console.log("Inserting");
+    const id = cassandra.types.Uuid.random();
+    // A Vector can be built from a plain Array (with an explicit subtype) or a Float32Array.
```
Comment (QuerthDP) [inline examples/vector/vector-insert-select.js]: Could you provide both version in example?

File: examples/vector/vector-search-ann.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        await new Promise((resolve) => setTimeout(resolve, 250));
```
Comment (QuerthDP) [inline examples/vector/vector-search-ann.js:100]: Is it just a sleep in JS?

**Review by QuerthDP — APPROVED** [review]
Looks fine to me.
Consider adding both vector representations in working with vector example.

Consider checking the index availability via an  `/api/v1/indexes/{keyspace}/{index}/status` Vector Store endpoint instead of retrying the SELECT.

File: examples/vector/vector-search-ann.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hmarszalek) [reply examples/vector/vector-search-ann.js]: Good catch — the retry was swallowing genuine schema/auth/connection failures.

On filtering without string parsing: there is no way to do it today. These all arrive as a napi `GenericFailure` (`src/errors.rs:178`) with a flat message, so a "still building the index" failure and a real CQL error are indistinguishable to the client — there is no structured server error code to switch on.

So instead of filtering, nothing is discarded: the last error is kept and attached as the `cause` of the timeout error.

```
There was an error Error: Vector search did not become available - is the Vector Store service running?
  [cause]: [Error [ExecutionError]: Database returned an error: ... Vector Store service is unavailable] {
    code: 'GenericFailure'
  }
```

File: examples/vector/vector-insert-select.js (hunk 3 above)
Comment (hmarszalek) [reply examples/vector/vector-insert-select.js]: Done — the example now inserts and reads back both forms: a `Vector` built from a plain `Array` with an explicit `"float"` subtype, and a `Float32Array`.

```
Inserting
Retrieved comment: "Great product, fast shipping!"
Retrieved vector: [0.11999999731779099,0.3400000035762787,...]
Retrieved comment: "Exactly as described"
Retrieved vector: [0.10999999940395355,0.2199999988079071,...]
```

File: examples/vector/vector-search-ann.js (hunk 4 above)
Comment (hmarszalek) [reply examples/vector/vector-search-ann.js]: Yes — `new Promise((resolve) => setTimeout(resolve, ms))` is the usual way to sleep in JS, since there is no built-in blocking sleep.

Node ships a promise-based version though, so I switched to it:

```js
const { setTimeout: sleep } = require("node:timers/promises");
...
await sleep(250);
```

**Comment by wprzytula** [comment]
> `examples/vector/vector-search-ann.js` stores comment embeddings in a `vector<float, 3>` column, creates a vector index, and runs an ANN similarity query. To make this runnable in CI, `.github/docker-compose.yml` now also starts the Vector Store indexing service and points the nodes at it with `--vector-store-primary-uri`.

What's the impact of this on CI's execution time?

**Comment by wprzytula** [comment]
Isn't this a blocker for considering vector type properly supported?
https://github.com/scylladb/nodejs-rs-driver/issues/380

**Comment by wprzytula** [comment]
Rust Driver includes a full integration test for vector type with all possible cql types inside: https://github.com/scylladb/scylla-rust-driver/blob/2c8539fc3c65b018d98630b4e8a137208070c303/scylla/tests/integration/types/cql_collections.rs#L608
Can you port it to Node.js RS Driver too? We need to verify all corner cases, because the vector type has weird and confusing de/ser, which might be broken in the JS code inherited from the old DSx driver.

**Review by wprzytula — CHANGES_REQUESTED** [review]
What you added looks good. It might be not enough though - see my other comments.

**Comment by Akvear** [comment]
> Rust Driver includes a full integration test for vector type with all possible cql types inside

Both C# drivers also test all types nested in a vector:
https://github.com/scylladb/csharp-driver/blob/36ded54dedbe6e3a023ca0178d91368f12923b85/src/Cassandra.IntegrationTests/Core/TypeSerializersTests.cs#L313

**Comment by hmarszalek** [comment]
> What's the impact of this on CI's execution time? 

The time needed for the examples job to run went from 1:40 to around 3:40.

**Comment by wprzytula** [comment]
> > What's the impact of this on CI's execution time?
> 
> The time needed for the examples job to run went from 1:40 to around 3:40.

I don't find such big increase acceptable for just one example, not strictly related to driver's functionality (just ScyllaDB itself with ANN queries).
Let's either not have this example (worse) or disable running this example in the CI (better).

**Comment by hmarszalek** [comment]
> Isn't this a blocker for considering vector type properly supported? #380

Vectors with primitive types and collections, and vector nested inside list/set pass just fine. What breaks is any parameterized type inside map or tuple.

**Comment by wprzytula** [comment]
> > Isn't this a blocker for considering vector type properly supported? #380
> 
> Vectors with primitive types and collections, and vector nested inside list/set pass just fine. What breaks is any parameterized type inside map or tuple.

We need to fix this as part of our effort to support vector type.

**Comment by hmarszalek** [comment]
> > > Isn't this a blocker for considering vector type properly supported? #380
> > 
> > 
> > Vectors with primitive types and collections, and vector nested inside list/set pass just fine. What breaks is any parameterized type inside map or tuple.
> 
> We need to fix this as part of our effort to support vector type.

Addressed in #521.

File: test/test-helper.js [hunk 5]
```diff
@@ -1872,8 +1870,8 @@ const dataProviderWithCollections = dataProvider
                 ],
                 customTypeName: "vector",
             },
-            value: data.value.map((value) => [value, value, value]),
-        }, */
+            value: data.value.map((value) => [value]),
```
Comment (wprzytula) [inline test/test-helper.js:1895]: commit `Enable vector<map> and vector<set> in dataProviderWithCollections`:
❓What's _the provider_? Why was it put like this in the code (duplicated thrice) - why did _the provider_ build it this way?

> The provider built each set as [value, value, value], but sets drop
> duplicates — so the server returned one element while the test expected
> three. I changed it to [value]: one element per set, nothing to
> deduplicate.

File: test/test-helper.js [hunk 6]
```diff
@@ -1637,6 +1637,28 @@ const dataProvider = [
...
+            code: types.dataTypes.custom,
+            customTypeName: "vector",
+            info: [{ code: types.dataTypes.varint }, 3],
+        },
+        value: [
+            types.Integer.fromString("-1"),
+            types.Integer.fromString("0"),
+            types.Integer.fromString("12345678901234567890"),
+        ],
+    },
```
Comment (wprzytula) [inline test/test-helper.js:1661]: ❓ Any idea why these two were missing from _primitives_?
❓ Are now all primitives included? What about decimal?

Comment (hmarszalek) [reply test/test-helper.js:1661]: > ❓ Any idea why these two were missing from primitives?

Probably an oversight :woman_shrugging: .

> ❓ Are now all primitives included? What about decimal?

The following types ARE present (I think that's all of them):
"ascii", "bigint", "blob", "boolean", "counter", "date", "decimal", "double", "duration", "float", "inet", "int", "smallint", "text", "time", "timestamp", "timeuuid", "tinyint", "uuid", "varchar", "varint".

Comment (wprzytula) [reply test/test-helper.js:1661]: I confirm these are all of them.

File: test/test-helper.js (hunk 5 above)
Comment (hmarszalek) [reply test/test-helper.js:1895]: > What's the provider?

`dataProvider` (in test-helper.js) lists one entry per CQL primitive: its name, the type info for vector<subtype, 3>, and three sample values. `dataProviderWithCollections` expands each of those into extra entries – list<T>, set<T>, map<int, T>, tuple<T, T>, vector<T, 3> and my_udt.

> Why was it put like this in the code (duplicated thrice) - why did the provider build it this way?

Dunno, but it was incorrect. The bug wasn't caught earlier because the tests weren't enabled.

**Review by wprzytula — APPROVED** [review]

**Review by wprzytula — APPROVED** [review]
